### PR TITLE
Release notes and migration guide for 17

### LINKED ISSUE #1392: Release notes for Keycloak 17
### Description

Release notes for Keycloak 17 has to contain information around the switch to Quarkus as the default distribution.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### LINKED ISSUE #1393: Migration from Keycloak.X preview
### Description

The migration guide for Keycloak 17 [1] should be updated to add steps to upgrade from Keycloak.X preview distribution to new default Quarkus distribution.

[1] changes-17_0_0.adoc in https://github.com/keycloak/keycloak-documentation/tree/main/upgrading/topics/keycloak

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by stianst**
Resolves #1393, resolves #1392

### CODE REVIEW

File: release_notes/topics/17_0_0.adoc [hunk 1]
```diff
@@ -4,22 +4,30 @@
...
+
+A lot of effort went into polishing and improving the Quarkus distribution to make it as good as an experience as possible. A few highlights include:
+
+* A new approach to documentation in form of server guides to help you install and configure Keycloak
+* Upgraded Quarkus to 2.7.0.Final
+* Configuration file is on longer Java specific, and aligns configuration keys with CLI arguments
+* Clearer separation between `build options` and `runtime configuration`.
+* `h2-mem` and `h2-file` databases renamed to `dev-mem` and `dev-file`.
+* Simplified enabling and disabling features
+* Custom, and unsupported, Quarkus configuration is done through `conf/quarkus.properties`.
```
Comment (DGuhr) [inline release_notes/topics/17_0_0.adoc]:
```suggestion
* Ability to add custom Java Options via JAVA_OPTS_APPEND (thx "dasniko")
* Initial logging capabilities
```

I think these are also worth mentioning, at least the logging one. wdyt?

File: upgrading/topics/keycloak/changes-17_0_0.adoc [hunk 2]
```diff
@@ -1,3 +1,27 @@
+= Default distribution is now powered by Quarkus
+
+The default distribution of Keycloak is now powered by Quarkus, which brings a number of breaking changes to you configure Keycloak and deploy custom providers. For more information check out the migration/migrating-to-quarkus[Quarkus Migration Guide].
+
+The WildFly distribution of Keycloak is now deprecated, with support ending June 2022. We recommend migrating to the Quarkus distribution as soon as possible. However, if you need to remain on the legacy WildFly distribution for some time, there's some changes:
```
Comment (DGuhr) [inline upgrading/topics/keycloak/changes-17_0_0.adoc]:
```suggestion
The WildFly distribution of Keycloak is now deprecated, with support ending June 2022. We recommend migrating to the Quarkus distribution as soon as possible. However, if you need to remain on the legacy WildFly distribution for some time, there are some changes to consider:
```

File: upgrading/topics/keycloak/changes-17_0_0.adoc [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+
+* Container now published to `quay.io/keycloak/keycloak:latest` and `quay.io/keycloak/keycloak:17.0.0`
+* Download on website renamed to `keycloak-17.0.0.[zip|tar.gz]`.
+* `conf/keycloak.properties` changed to `conf/keycloak.conf`, which unifies configuration keys between the config file and CLI arguments.
+* Clearer separation between `build options` and `runtime configuration`.
+* Custom Quarkus configuration is done through `conf/quarkus.properties`.
+* `h2-mem` and `h2-file` databases renamed to `dev-mem` and `dev-file`.
+* Features are now enabled/disabled with `--features` and `--features-disabled` replacing the previous approach that had an separate config key for each feature.
+* Runtime configuration can no longer be passed to `kc.[sh|bat] build` and is no longer persisted in the build
+
```
Comment (DGuhr) [inline upgrading/topics/keycloak/changes-17_0_0.adoc]:
```suggestion
* Logging does not need to be configured using raw quarkus properties anymore, you can use `--log-level` and `--log-format` now. 
```

File: release_notes/topics/17_0_0.adoc [hunk 4]
```diff
@@ -4,22 +4,31 @@
...
+* Ability to add custom Java Options via JAVA_OPTS_APPEND (thx "dasniko")
```
Comment (stianst) [inline release_notes/topics/17_0_0.adoc]:
```suggestion
* Ability to add custom Java Options via JAVA_OPTS_APPEND (thanks to https://github.com/dasniko[dasniko])
```

File: upgrading/topics/keycloak/changes-17_0_0.adoc [hunk 5]
```diff
@@ -1,3 +1,27 @@
...
+* Container now published to `quay.io/keycloak/keycloak:latest` and `quay.io/keycloak/keycloak:17.0.0`
+* Download on website renamed to `keycloak-17.0.0.[zip|tar.gz]`.
+* `conf/keycloak.properties` changed to `conf/keycloak.conf`, which unifies configuration keys between the config file and CLI arguments.
+* Clearer separation between `build options` and `runtime configuration`.
+* Custom Quarkus configuration is done through `conf/quarkus.properties`.
+* `h2-mem` and `h2-file` databases renamed to `dev-mem` and `dev-file`.
+* Features are now enabled/disabled with `--features` and `--features-disabled` replacing the previous approach that had an separate config key for each feature.
+* Runtime configuration can no longer be passed to `kc.[sh|bat] build` and is no longer persisted in the build
+* Logging does not need to be configured using raw quarkus properties anymore, you can use `--log-level` and `--log-format` now.
```
Comment (stianst) [inline upgrading/topics/keycloak/changes-17_0_0.adoc]:
```suggestion
* Logging level and format is now configured with `--log-level` and `--log-format`, while in the past these had to be configured using unsupported Quarkus properties.
```

**Review by DGuhr — DISMISSED** [review]

File: release_notes/topics/17_0_0.adoc [hunk 6]
```diff
@@ -4,22 +4,31 @@
...
+
+* A new approach to documentation in form of server guides to help you install and configure Keycloak
+* Upgraded Quarkus to 2.7.0.Final
+* Configuration file is on longer Java specific, and aligns configuration keys with CLI arguments
+* Clearer separation between `build options` and `runtime configuration`.
+* `h2-mem` and `h2-file` databases renamed to `dev-mem` and `dev-file`.
+* Simplified enabling and disabling features
+* Custom, and unsupported, Quarkus configuration is done through `conf/quarkus.properties`.
+* Ability to add custom Java Options via JAVA_OPTS_APPEND (thanks to https://github.com/dasniko[dasniko])
+* Initial logging capabilities
```
Comment (pedroigor) [inline release_notes/topics/17_0_0.adoc:23]:
```suggestion
* Initial logging capabilities
* Initial support for Cross-DC
* User-defined profiles are no longer supported but using different configuration files to achieve the same goal
* Quickstarts updated to use the new distribution
```

**Review by pedroigor — DISMISSED** [review]

**Pull request merged**